nssp/delphi_nssp/run.py

```diff
@@ -72,18 +70,28 @@ def run_module(params):
         Nested dictionary of parameters.
     """
     start_time = time.time()
-    logger = get_structured_logger(
-        __name__,
-        filename=params["common"].get("log_filename"),
-        log_exceptions=params["common"].get("log_exceptions", True),
-    )
+    custom_run = params["common"].get("custom_run", False)
+    # logger doesn't exist yet means run_module is called from normal indicator run (instead of patching)
+    if not logger:
+        logger = get_structured_logger(
+            __name__,
+            filename=params["common"].get("log_filename"),
+            log_exceptions=params["common"].get("log_exceptions", True),
+        )
+        if custom_run:
+            logger.warning("custom_run flag is on despite direct indicator run call. Normal indicator run continues.")
+            custom_run = False
     export_dir = params["common"]["export_dir"]
     socrata_token = params["indicator"]["socrata_token"]
-
     run_stats = []
     ## build the base version of the signal at the most detailed geo level you can get.
     ## compute stuff here or farm out to another function or file
-    df_pull = pull_nssp_data(socrata_token)
+    if custom_run:
+        issue_date = params.get("patch", {}).get("current_issue", None)
```

Comment:
I think we want to fail if the issue date or the current issue doesn't exist right? because if we explicitly set the flag to be a custom run, but if current issue is None, then it's going to run a normal run...which would make things confusing since the logger would be for patching, but still doing a regular run.

Reply:
Furthermore if you plan to kepp the patch param validation; you already did the check ahead of time so it should be fine to change it to the [] operator instead of get

Reply:
If we're in normal patching behavior, which means:
1.`custom_run` is True 
2.`patch.py` was called to do patching
but 
3. `current_issue` is None 
Then the check for current issue being none is already done [here](https://github.com/cmu-delphi/covidcast-indicators/pull/2000/files#diff-8fce7ae14a0e2097dd3da98817cd6258af5a7596b2a7117339da5ce8abffc875R118-R121), so the code won't get to `run.py` in the first place if there's no current issue. It will just skip to the next day's data.
In the above case, logger will be from patching, as [this](https://github.com/cmu-delphi/covidcast-indicators/pull/2000/files#diff-0f52ac57508ccf1c326cb006bdbc426344d14d0dcdbc79838a67fa0f6999ff90R75-R83) block won't get triggered

If we're doing cursed behavior:
1.`custom_run` is True 
but
2. `run.py` was called instead of patch.py
Then [this chunk](https://github.com/cmu-delphi/covidcast-indicators/pull/2000/files#diff-0f52ac57508ccf1c326cb006bdbc426344d14d0dcdbc79838a67fa0f6999ff90R81-R83) will stop the cursed run and assume this is a normal run, set the `custom_run` flag to false automatically, and call `pull_nssp_data(socrata_token)` with no issue date or source dir, which means we're pulling the latest data. 
The logger will be normal run in this case, because [this](https://github.com/cmu-delphi/covidcast-indicators/pull/2000/files#diff-0f52ac57508ccf1c326cb006bdbc426344d14d0dcdbc79838a67fa0f6999ff90R75-R83) block will get triggered.

Reply:
So you intentionally want the behavior for the "cursed" run to run as a normal run? I think if that's the case, you should have make a comment when you change the custom run flag.

Reply:
Okay I see that you were intentionally setting convert to regular run if it's not a patch. If that's the case, then this is fine

Reply:
Ah yup; you are right about the logger! I forgot that it's passed in...

That was a bad example...I still think it's better to explicitly use the [] where it's clear that if the custom flag is true; we expect the issue and the source dir to exist. I was trying to give a weird behavior where it would be confusing/unclear but 🙃

Reply:
I don't particularly care how we handle this and [the `custom_flag`](https://github.com/cmu-delphi/covidcast-indicators/pull/2000/files#r1718538840), but this logic should be the same between all of our patch modules. Please consult how we do this in the `google-symptoms` and `doctor-visits` patchers.